projects/packages/my-jetpack/src/products/class-stats.php

```diff
@@ -145,21 +160,12 @@ public static function get_wpcom_free_product_slug() {
 	 * @return boolean
 	 */
 	public static function has_required_plan() {
-		$purchases_data = Wpcom_Products::get_site_current_purchases();
-		if ( is_wp_error( $purchases_data ) ) {
+		// The only requirement for Stats is a connection.
+		$connection = new Connection_Manager();
+		if ( ! $connection->is_connected() || ! $connection->is_user_connected() ) {
```

Comment:
Hmmm, this connection check seems out of place in the `has_required_plan` function 🤔 

I wonder if we should add another static property on the product class called `requires_site_connection`. If the product requires a site connection and the site is not connected, I think we could have the `get_status` function return "inactive".

We could probably remove this function with a change like that.

What do you think?

Reply:
Yes that sounds like a good solution, thanks @jboland88!  Although the 'detailed' Stats page currently will not show/render (is blocked) for users without a user-connection. So I'll need to make sure we can open it to users without a user-connection. I'll check if I can accomplished this fairly easily. 👍

Reply:
> Stats page currently will not show/render (is blocked) for users without a user-connection. So I'll need to make sure we can open it to users without a user-connection. I'll check if I can accomplished this fairly easily. 👍

Maybe we can still link there with the "Show detailed stats" CTA for now and allow the user to connect from that page if they are not currently connected? I think changing the connection required for that page is out of scope for this PR.

Reply:
@jboland88, Yes I agree changing the connection required for Stats is out of scope for this PR.
The I made it work now is: If there is no connection, the card show 'Inactive' with a "Learn more" button which points to the Stats product interstitial (Same behavior as the other products), which when clicked will lead to the "connect after checkout" flow with Stats commercial in the cart.
As for the `has_required_plan()` function, I removed the connection check, but instead simply just return `true` because a purchased plan is not currently a requirement for Stats. Stats functionality works right out of the box.
What do you think?